### DISCUSSION TITLE
How to fall back to stable local AP mode if no Wifi networks found?

### DISCUSSION THREAD

**Discussion by HankLloydRight** [Hieromon/AutoConnect#378] [How can I?]
Hello,
  Like others, I'm amazed at the library you have created here. Thank you!

 I've been trying for hours to try to get this working on ESP32.

Here is what I'd like to happen, if possible, but I can't find the right combination of settings/options to make it work like this:

1. Power up ESP32
2. AutoConnect looks for previous APs to connect to, and if any are found, it connects and Portal.Begin() ends and loop() begins.

(Note -- it works great up to step 2 -- I can't get Step 3, below to work)

3. After a timeout period (say 30 seconds), if no previous Wifi APs are found to connect to, create a local AP with the provided credentials, assign a local IP to the ESP32 (172.217.x.x), then Portal.Begin() ends and loop() begins.

So after 30 seconds, the device is either connected to a Wifi network, OR it creates its own network. 

If I set portalTimeout, then the local AP goes away.  If I set immediateStart, the loop() starts but with no way to connect to the ESP32 (no network or IP address to connect to). 

I'm sure this is possible, but I just can't figure it out! Thank you!!

**Comment by Hieromon** [comment]
@HankLloydRight The reply has been delayed. I don't have much time because I give priority to the correspondence between ESP8266 3.0.0 and ESP32 2.0.0.

Have you tried the combination of portalTimeout and [retainPortal](https://hieromon.github.io/AutoConnect/apiconfig.html#retainportal)?
Also, [**Inside AutoConnect::begin**](https://hieromon.github.io/AutoConnect/lsbegin.html) in the documentation may help you.

**Comment by HankLloydRight** [comment]
Thanks for your reply. I had not found the retainPortal option before, I will definitely try that out and report back.
